model/skipgraph/identifier.go

```diff
@@ -56,15 +81,17 @@ func (i Identifier) Compare(other Identifier) Comparison {
 		cmp := bytes.Compare(i[index:index+1], other[index:index+1])
 		switch cmp {
 		case 1:
-			return Comparison{CompareGreater, DebugInfo(i, other, CompareGreater, index), uint32(index)}
+			cr, _ := NewComparisonResult(CompareGreater)
```

Comment:
As a best practice, let's alway check for the returned error. In this case, we should bubble it up.

Reply:
Please see the updated version in this commit 789dbd55471e216a657254c5ca2e69390f1cf938